### PR TITLE
fix: support setting agent username and group in user APIs.

### PR DESCRIPTION
**Pull request by ioga**
## Description

grpc-style user APIs were missing support for username / groupname fields in `AgentUserGroups`, i.e. `link-with-agent-user` feature. This is not an externally facing fix: `AgentUserGroups` isn't currently in use by WebUI, CLI still uses the old style.

Under the hood, user name and group name are NOT NULL fields in the table. Prior to this fix, they'd be reset to empty strings due to protobuf / golang empty struct initializations.

Technically, uid and gid are critical for the system, while user name and group name are cosmetic, however if we enforce them in one place (db and `link-with-agent-user`) it probably makes sense to enforce them everywhere.

## Test Plan

Expanded automated testing.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

File: master/internal/api_user.go [hunk 1]
```diff
@@ -334,8 +344,10 @@ func (a *apiServer) PatchUser(
 	var ug *model.AgentUserGroup
 	if pug := req.User.AgentUserGroup; pug != nil {
 		ug = &model.AgentUserGroup{
-			UID: int(req.User.AgentUserGroup.AgentUid),
-			GID: int(req.User.AgentUserGroup.AgentGid),
+			UID:   int(req.User.AgentUserGroup.AgentUid),
+			GID:   int(req.User.AgentUserGroup.AgentGid),
+			User:  req.User.AgentUserGroup.AgentUser,
+			Group: req.User.AgentUserGroup.AgentGroup,
```
Comment (mapmeld) [inline master/internal/api_user.go:350]: is there any restriction  or can a user make an agent-user or agent-group association for anyone?

File: harness/determined/common/api/bindings.py [hunk 2]
```diff
@@ -478,22 +478,30 @@ def __init__(
         self,
         *,
         agentGid: "typing.Optional[int]" = None,
+        agentGroup: "typing.Optional[str]" = None,
```
Comment (NicholasBlaskey) [inline harness/determined/common/api/bindings.py:481]: should we switch the CLI set agent user group over with this change? 

Waiting to do it at all once or bit by bit as we make sure functionality is matched both seem fine to me but curious on your thoughts.

**Review by NicholasBlaskey — APPROVED** [review]
lgtm

File: master/internal/api_user.go (hunk 1 above)
Comment (mapmeld) [reply master/internal/api_user.go:350]: :+1:

File: harness/determined/common/api/bindings.py (hunk 2 above)
Comment (ioga) [reply harness/determined/common/api/bindings.py:481]: this is a judgement call: on one hand, you are right in advocating this. on the other, I am working from a mindset of a critical path towards storage solution, and we basically have this work  outlined and ticketed as one of our four top tech debt items.

**Comment by mapmeld** [comment]
This is last-minute but could it be a userId instead of agentUsername? We were just discussing features to make changing username easier and this would affect username here

**Pull request merged**